internal/controller/unmanagedcluster_controller.go

```diff
@@ -0,0 +1,505 @@
+				Labels:    map[string]string{v1beta1.GroupVersion.Identifier(): hmc.GroupVersion.Version, v1beta1.ClusterNameLabel: unmanagedCluster.Name},
+			},
+			Spec: v1beta1.MachineSpec{
+				ClusterName: unmanagedCluster.Name,
+				Bootstrap: v1beta1.Bootstrap{
+					DataSecretName: &kubeConfigSecretName,
+				},
+				InfrastructureRef: corev1.ObjectReference{
+					Kind:       "UnmanagedMachine",
+					Namespace:  unmanagedCluster.Namespace,
+					Name:       node.Name,
+					APIVersion: hmc.GroupVersion.Identifier(),
+				},
+				ProviderID: &node.Spec.ProviderID,
+			},
+			Status: v1beta1.MachineStatus{
+				NodeRef: &corev1.ObjectReference{
+					Kind:       "Node",
+					Name:       node.Name,
+					APIVersion: "v1",
+				},
+				NodeInfo:               &corev1.NodeSystemInfo{},
+				CertificatesExpiryDate: nil,
+				BootstrapReady:         true,
+				InfrastructureReady:    true,
+			},
+		}
+
+		if _, ok := node.Labels[v1beta1.NodeRoleLabelPrefix+"/control-plane"]; ok {
+			if machine.Labels == nil {
+				machine.Labels = make(map[string]string)
+			}
+			machine.Labels[v1beta1.MachineControlPlaneLabel] = "true"
+		}
+		err = r.Create(ctx, &machine)
+		if err != nil && !apierrors.IsAlreadyExists(err) {
+			return fmt.Errorf("failed to create machine: %w", err)
+		}
+	}
+
+	// cleanup any orphaned unmanaged machines and capi machines
+	for _, existingUnmanagedMachine := range existingMachinesByName {
+		if err := r.Delete(ctx, existingUnmanagedMachine); err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to delete orphaned unmanaged machine: %w", err)
+		}
+
+		if err := r.Delete(ctx, &v1beta1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      existingUnmanagedMachine.Name,
+				Namespace: unmanagedCluster.Namespace,
+			},
+		}); err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to delete orphaned machine: %w", err)
+		}
+	}
+	return nil
+}
+
+func (r *UnmanagedClusterReconciler) getNodeList(ctx context.Context, unmanagedCluster *hmc.UnmanagedCluster) (*corev1.NodeList, error) {
+	l := ctrl.LoggerFrom(ctx)
+	clusterClient, err := clusterproxy.GetCAPIKubernetesClient(ctx, l, r.Client, r.Client.Scheme(), unmanagedCluster.Namespace, unmanagedCluster.Name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to remote cluster: %w", err)
+	}
+
+	nodelist := &corev1.NodeList{}
+	if err := clusterClient.List(ctx, nodelist); err != nil {
+		return nil, fmt.Errorf("failed to list cluster nodes: %w", err)
+	}
+	return nodelist, nil
+}
+
+func (r *UnmanagedClusterReconciler) reconcileUnmanagedCluster(ctx context.Context, unmanagedCluster *hmc.UnmanagedCluster) (ctrl.Result, error) {
+	l := ctrl.LoggerFrom(ctx)
+	if !unmanagedCluster.DeletionTimestamp.IsZero() {
+		l.Info("Deleting UnmanagedCluster")
+		return r.reconcileDeletion(ctx, unmanagedCluster)
+	}
+
+	if err := r.createCluster(ctx, unmanagedCluster); err != nil {
+		return ctrl.Result{RequeueAfter: DefaultRequeueInterval}, err
```

Comment:
I suggest to rely on exponential backoff instead of a fixed one in general (unless there are reasons for that)